tensorflow/python/eager/backprop.py

```diff
@@ -938,6 +957,9 @@ def gradient(self,
             "gradient in order to compute higher order "
             "derivatives.", 1)
 
+    if check_float_dtype(target) == False:
```

Comment:
Same comment as above: fuse into the loop and make the error message more specific.

Reply:
Also add a similar check to `sources` below.

Reply:
Done.